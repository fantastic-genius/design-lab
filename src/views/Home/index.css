.home-content .history-action {
  width: 6.4rem;
  height: 12.8rem;
  background: var(--main-white-color);
  border-radius: 0.8rem;
  font-size: 1.3rem;
}

.home-content .part-container {
  width: 12rem;
  height: 40rem;
  border-radius: 1.2rem;
  background: var(--main-white-color);
}

.home-content .part-item {
  width: 12rem;
  height: 10rem;
}

.home-content .part-item-active {
  border-radius: 1.2rem;
  border: solid 1px var(--main-primary-color);
}

.home-content .zoom-con {
  width: 4.8rem;
  height: 4.8rem;
  border-radius: 0.6rem;
  background: var(--main-white-color);
}

.home-content .shirt-con {
  width: 560px;
  height: calc(100vh - 140px);
}

.home-content .shirt-img {
  width: auto;
  height: calc(100vh - 140px);
}
