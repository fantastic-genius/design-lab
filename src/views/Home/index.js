import './index.css';
import React, { useState, useCallback, useRef, useEffect } from 'react';
import { Layout } from 'antd';
import { Stage, Layer, Image, Rect } from 'react-konva';
import useImage from 'use-image';
import { customAlphabet } from 'nanoid';
import { alphanumeric } from 'nanoid-dictionary'

//components
import {
  SideBar,
  AppLayout,
  AddText,
  QuickAction,
  UploadDesign,
  FontSection,
  CustomImage,
  CustomText
} from '../../components';

//icons
import {
  Undo,
  Redo,
  ShirtFront,
  ShirtBack,
  RightSleeve,
  LeftSleeve,
  ZoomIn,
  ZoomOut
} from '../../icons';

//images
import Shirt from '../../assets/img/Shirt.png';

const Home = () => {
  const { Content } = Layout;
  const [selectedMenu, setSelectedMenu] = useState(null);
  const stageContainerRef = useRef();
  const [stageDimension, setStageDimention] = useState({ width: 0, height: 0 });
  const [addedItems, setAddedItems] = useState([]);
  const [currentFont, setCurrentFont] = useState('Roboto');
  const [selectedTextItem, setSelectedTextItem] = useState(null);
  const [selectedImageItem, setSelectedImageItem] = useState(null);
  const [isTextEditEnabled, setIsTextEditEnabled] = useState(false);
  const designAreaConfig = {
    width: 200,
    height: 280
  };
  

  const onCloseMenu = useCallback(() => {
    if(isTextEditEnabled && selectedMenu === 'text'){
      setIsTextEditEnabled(false);
      setSelectedTextItem(null);
    }
    setSelectedMenu(null);
  },[isTextEditEnabled, selectedMenu]);

  //function to add text
  const handleAddText = useCallback((text) => {
    const nanoid = customAlphabet(alphanumeric, 12);
    const id = nanoid();
    const textNode = {
      id,
      text,
      x: 0,
      y: addedItems.length ?  Math.random() * designAreaConfig.height : 0,
      type: 'text',
      font: currentFont,
      fontSize: 36
    }

    setAddedItems(prev => ([...prev, textNode]));
  }, [addedItems, designAreaConfig.height, currentFont]);


  //function to edit text and text font
  const handleEditText = useCallback((textItem, newTextValue=null) => {
    const newTextItems = addedItems.map(item => {
      if(item.id === textItem.id){
        return {
          ...item,
          font: currentFont,
          ...(newTextValue && { text: newTextValue })
        }
      }
      return item;
    });
    setAddedItems(newTextItems);
  }, [addedItems, currentFont]);

  //function to add image
  const handleAddImage = useCallback((image, height, width) => {
    let newWidth = width;
    let newHeight = height;

    if(width > designAreaConfig.width){
      newWidth = designAreaConfig.width;
      newHeight = (designAreaConfig.width * height)/width;
    }
    const nanoid = customAlphabet(alphanumeric, 12);
    const id = nanoid();
    const imageNode = {
      id,
      image,
      x: 0,
      y: addedItems.length ?  Math.random() * designAreaConfig.height : 0,
      type: 'image',
      height: newHeight,
      width: newWidth
    }

    setAddedItems(prev => ([...prev, imageNode]));
  }, [addedItems, designAreaConfig]);

  //funtion to change font
  const handleAddFont = useCallback((font) => {
    setCurrentFont(font);
  }, []);

  //function to deselect any selected item
  const clearSelection = useCallback((type) => {
    if(selectedImageItem) setSelectedImageItem(null);
    if(selectedTextItem) {
      setSelectedTextItem(null);
      setIsTextEditEnabled(false);
      setSelectedMenu(null);
    }
  }, [selectedTextItem, selectedImageItem]);

  //function to remove any added items (text or image)
  const handleRemove = useCallback((id) => {
    const newItem = addedItems.filter(item => item.id !== id);
    setAddedItems(newItem);
    clearSelection();
  },[addedItems, clearSelection])

  const onGoBackToText = useCallback(() => setSelectedMenu('text'), []);

  const onGoToFont = useCallback(() => setSelectedMenu('font'), []);

  //function to contron lenereing of slected menu content
  const renderMenuContent = useCallback(() => {
    switch (selectedMenu) {
      case 'text':
        return <AddText
          closeMenu={onCloseMenu}
          handleAddText={handleAddText}
          handleChangeFont={onGoToFont}
          selectedFont={currentFont}
          selectedText={selectedTextItem}
          editMode={isTextEditEnabled}
          handleEditText={handleEditText}
        />;
      case 'upload':
          return <UploadDesign closeMenu={onCloseMenu} handleAddImage={handleAddImage} />;
      case 'font':
        return (
          <FontSection
            closeMenu={onCloseMenu}
            handleAddFont={handleAddFont}
            onGoBack={onGoBackToText}
            selectedText={selectedTextItem?.text || ''}
          />
        );
      default:
        return <QuickAction setSelectedMenu={setSelectedMenu} />;
    }
  }, [
    selectedMenu,
    handleAddText,
    onCloseMenu,
    handleAddImage,
    handleAddFont,
    onGoBackToText,
    onGoToFont,
    currentFont,
    selectedTextItem,
    isTextEditEnabled,
    handleEditText
  ]); 

  const ShirtImage = ({ src }) => {
    const [image] = useImage(src);

    return <Image image={image} height={stageDimension.height} width={stageDimension.height * 0.76} />
  }

  //function to be called when an added item is tapped or clicked
  const onSelectItem = useCallback((item) => {
    if(item.type === 'text'){
      if(selectedImageItem) setSelectedImageItem(null);
      setSelectedTextItem(item);
      setIsTextEditEnabled(true);
      if(selectedMenu !== 'text'){
        setSelectedMenu('text');
      }
    }else if(item.type === 'image'){
      if(selectedTextItem) {
        setSelectedTextItem(null);
        setIsTextEditEnabled(false);
        setSelectedMenu(null);
      }
      setSelectedImageItem(item);
    }
  }, [selectedMenu, selectedImageItem, selectedTextItem])

  //function to change the attribute of an item during scaling
  const handleAttributeChange = useCallback((newAttrs, i) => {
    const newItems = addedItems.slice();
    newItems[i] = newAttrs;
    setAddedItems(newItems);
  }, [addedItems])

  const renderItem = useCallback((item, i) => {
    const { id, type, text, x, y, width, height, image, font, fontSize , rotation } = item;
    switch (type) {
      case 'text':
        return (
          <CustomText
            key={id}
            text={text}
            x={x}
            y={y}
            draggable
            fontSize={fontSize}
            wrap='char'
            width={width}
            height={height}
            rotation={rotation}
            fontFamily={font}
            onClick={() => onSelectItem(item)}
            onTap={() => onSelectItem(item)}
            shapeProps={item}
            onChange={(newAttrs) => handleAttributeChange(newAttrs, i)}
            isSelected={selectedTextItem && selectedTextItem.id === id}
            designAreaConfig={designAreaConfig}
            onRemove={() => handleRemove(id)}
          />
        );
      case 'image':
        return (
          <CustomImage
            key={id}
            src={image}
            width={width}
            height={height}
            x={x}
            y={y}
            rotation={rotation}
            shapeProps={item}
            onClick={() => onSelectItem(item)}
            onTap={() => onSelectItem(item)}
            onChange={(newAttrs) => handleAttributeChange(newAttrs, i)}
            isSelected={selectedImageItem && selectedImageItem.id === id}
            designAreaConfig={designAreaConfig}
            onRemove={() => handleRemove(id)}
          />
        );
      default:
        break;
    }
  }, [
    selectedTextItem,
    selectedImageItem,
    handleAttributeChange,
    onSelectItem,
    handleRemove
  ]);

  //render the text and images added
  const renderNodes = useCallback(() => {
    const nodes = addedItems.map(renderItem);
    return nodes;
  }, [addedItems, renderItem]);

  useEffect(() => {
    if(stageContainerRef.current){
      setStageDimention({
        width: stageContainerRef.current.offsetWidth,
        height: stageContainerRef.current.offsetHeight
      })
    }
  }, [stageContainerRef]);

  useEffect(() => {
    if((selectedMenu === 'upload')  && isTextEditEnabled ){
      setIsTextEditEnabled(false);
      setSelectedTextItem(null);
    }
  }, [selectedMenu, isTextEditEnabled]);


  return (
    <AppLayout>
      <div className='d-flex flex-row home-page'>
        <SideBar setSelectedMenu={setSelectedMenu} selectedMenu={selectedMenu} />
        {renderMenuContent()}
        <Layout className='home-content dl-pt-12 dl-pb-12 dl-pl-12 dl-pr-12'>
          <Content className='d-flex'>
            <div className='history-action dl-pt-12 dl-pb-12 dl-pl-12 dl-pr-12 d-flex flex-column justify-content-between'>
              <div className='text-center dl-pointer'>
                <Undo />
                <p className='dl-text-gray-600 dl-font-weight-500 mt-2'>Undo</p>
              </div>
              <div className='text-center opacity-25'>
                <Redo />
                <p className='dl-text-gray-600 dl-font-weight-500 mt-2'>Redo</p>
              </div>
            </div>
            <div className='flex-grow-1 text-center'>
              <div className='shirt-con d-flex justify-content-center' ref={stageContainerRef}>
                <Stage id='design-stage' height={stageDimension.height} width={stageDimension.height * 0.76}>
                  <Layer>
                    <ShirtImage src={Shirt} />
                  </Layer>
                  <Layer
                    x={stageDimension.height * 0.190}
                    y={stageDimension.height * 0.162}
                    clipX={0}
                    clipY={0}
                    clipHeight={designAreaConfig.height}
                    clipWidth={designAreaConfig.width}
                    height={50}
                    width={80}
                    id='design-con'
                    onDblClick={clearSelection}
                  >
                    <Rect
                      x={0}
                      y={0}
                      strokeWidth={1}
                      width={designAreaConfig.width}
                      height={designAreaConfig.height}
                      stroke='#334E68'
                      dash={[2, 2]}
                    />
                    {renderNodes()}
                  </Layer>
                </Stage>
              </div>
            </div>
            <div>
              <div className='part-container'>
                <div className='part-item d-flex flex-column justify-content-center align-items-center part-item-active dl-pointer'>
                  <ShirtFront />
                  <p className='dl-text-gray-600 dl-font-weight-600 dl-text-small text-uppercase dl-mt-8'>Front</p>
                </div>
                <div className='part-item d-flex flex-column justify-content-center align-items-center dl-pointer'>
                  <ShirtBack />
                  <p className='dl-text-gray-600 dl-font-weight-600 dl-text-small text-uppercase dl-mt-8'>Back</p>
                </div>
                <div className='part-item d-flex flex-column justify-content-center align-items-center dl-pointer'>
                  <RightSleeve />
                  <p className='dl-text-gray-600 dl-font-weight-600 dl-text-small text-uppercase dl-mt-8'>Right Sleeve</p>
                </div>
                <div className='part-item d-flex flex-column justify-content-center align-items-center dl-pointer'>
                  <LeftSleeve />
                  <p className='dl-text-gray-600 dl-font-weight-600 dl-text-small text-uppercase dl-mt-8'>Left Sleeve</p>
                </div>
              </div>
              <div className="d-flex justify-content-between dl-mt-8">
                <div className="zoom-con d-flex justify-content-center align-items-center dl-pointer">
                  <ZoomOut />
                </div>
                <div className="zoom-con d-flex justify-content-center align-items-center dl-pointer">
                  <ZoomIn />
                </div>
              </div>
            </div>
          </Content>
      </Layout>
      </div>
    </AppLayout>
  );
}

export default Home;
