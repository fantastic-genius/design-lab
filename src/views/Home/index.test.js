import { render } from '@testing-library/react';
import Home from './index';

test('Layout renders properly', () => {
  const { container } = render(<Home />);
  expect(container).toMatchSnapshot();
});
