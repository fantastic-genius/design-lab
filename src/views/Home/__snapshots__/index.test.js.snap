// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders properly 1`] = `
<div>
  <section
    class="ant-layout layout-container w-100"
  >
    <div>
      <nav
        class="nav-con navbar navbar-expand-md navbar-light"
      >
        <a
          class="brand-name navbar-brand"
          data-testid="top-logo"
          href="/"
        >
          <img
            alt="SameDay Custom"
            src="Logo.svg"
          />
        </a>
        <button
          aria-label="Toggle navigation"
          class="navbar-toggler"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
        >
          <ul
            class="ml-auto dl-font-weight-500 dl-text-semi-small w-100 navbar-nav"
          >
            <li
              class="menu-item dl-text-gray-700 nav-item"
              data-testid="top-lmenu"
            >
              <a
                class="active-menu nav-link"
                href="#"
              >
                Design Lab
              </a>
            </li>
            <li
              class="menu-item dl-text-gray-700 nav-item"
              data-testid="top-lmenu"
            >
              <a
                class="nav-link"
                href="/investments"
              >
                Products
              </a>
            </li>
            <li
              class="menu-item dl-text-gray-700 nav-item"
              data-testid="top-lmenu"
            >
              <a
                class="nav-link"
                href="#"
              >
                How it works
              </a>
            </li>
            <li
              class="menu-item dl-text-gray-700 me-auto nav-item"
              data-testid="top-lmenu"
            >
              <a
                class="nav-link"
                href="#"
              >
                Bulk pricing
              </a>
            </li>
            <li
              class="nav-item"
              data-testid="top-rmenu"
            >
              <a
                class="get-help-btn nav-btn nav-link"
                href="#"
              >
                <svg
                  class="me-2"
                >
                  Chat.svg
                </svg>
                Get help
                <svg
                  class="ms-2"
                >
                  CaretDown.svg
                </svg>
              </a>
            </li>
            <li
              class="nav-item"
              data-testid="top-rmenu"
            >
              <a
                class="your-acct-btn nav-btn nav-link"
                href="#"
              >
                <svg
                  class="me-2"
                >
                  Person.svg
                </svg>
                 Your Account 
                <svg
                  class="ms-2"
                >
                  CaretDown.svg
                </svg>
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
    <section
      class="ant-layout ant-layout-has-sider"
    >
      <main
        class="ant-layout-content content"
      >
        <div
          class="d-flex flex-row home-page"
        >
          <aside
            class="ant-layout-sider ant-layout-sider-dark sidebar-component"
            style="flex: 0 0 100px; max-width: 100px; min-width: 100px; width: 100px;"
          >
            <div
              class="ant-layout-sider-children"
            >
              <ul
                class="ant-menu ant-menu-root ant-menu-inline ant-menu-light"
                data-menu-list="true"
                role="menu"
                tabindex="0"
              >
                <li
                  class="ant-menu-item ant-menu-item-only-child"
                  data-menu-id="rc-menu-uuid-test-text"
                  data-testid="side-menu"
                  role="menuitem"
                  style="padding-left: 24px;"
                  tabindex="-1"
                >
                  <span
                    class="ant-menu-title-content"
                  >
                    <div
                      class="text-center"
                    >
                      <div>
                        <svg
                          fill="white"
                        >
                          Text.svg
                        </svg>
                      </div>
                      <p
                        class="ml-2 px-3 dl-mt-12 text-wrap text-center"
                      >
                        Add Text
                      </p>
                    </div>
                  </span>
                </li>
                <li
                  class="ant-menu-item ant-menu-item-only-child"
                  data-menu-id="rc-menu-uuid-test-template"
                  data-testid="side-menu"
                  role="menuitem"
                  style="padding-left: 24px;"
                  tabindex="-1"
                >
                  <span
                    class="ant-menu-title-content"
                  >
                    <div
                      class="text-center"
                    >
                      <div>
                        <svg
                          fill="white"
                        >
                          Art.svg
                        </svg>
                      </div>
                      <p
                        class="ml-2 px-3 dl-mt-12 text-wrap text-center"
                      >
                        Use Template
                      </p>
                    </div>
                  </span>
                </li>
                <li
                  class="ant-menu-item ant-menu-item-only-child"
                  data-menu-id="rc-menu-uuid-test-upload"
                  data-testid="side-menu"
                  role="menuitem"
                  style="padding-left: 24px;"
                  tabindex="-1"
                >
                  <span
                    class="ant-menu-title-content"
                  >
                    <div
                      class="text-center"
                    >
                      <div>
                        <svg
                          fill="white"
                        >
                          Upload.svg
                        </svg>
                      </div>
                      <p
                        class="ml-2 px-3 dl-mt-12 text-wrap text-center"
                      >
                        Upload design
                      </p>
                    </div>
                  </span>
                </li>
                <li
                  class="ant-menu-item ant-menu-item-only-child"
                  data-menu-id="rc-menu-uuid-test-colors"
                  data-testid="side-menu"
                  role="menuitem"
                  style="padding-left: 24px;"
                  tabindex="-1"
                >
                  <span
                    class="ant-menu-title-content"
                  >
                    <div
                      class="text-center"
                    >
                      <div>
                        <svg
                          fill="white"
                        >
                          Color.svg
                        </svg>
                      </div>
                      <p
                        class="ml-2 px-3 dl-mt-12 text-wrap text-center"
                      >
                        Product Colors
                      </p>
                    </div>
                  </span>
                </li>
                <li
                  class="ant-menu-item ant-menu-item-only-child"
                  data-menu-id="rc-menu-uuid-test-team"
                  data-testid="side-menu"
                  role="menuitem"
                  style="padding-left: 24px;"
                  tabindex="-1"
                >
                  <span
                    class="ant-menu-title-content"
                  >
                    <div
                      class="text-center"
                    >
                      <div>
                        <svg
                          fill="white"
                        >
                          TeamNames.svg
                        </svg>
                      </div>
                      <p
                        class="ml-2 px-3 dl-mt-12 text-wrap text-center"
                      >
                        Add Team names
                      </p>
                    </div>
                  </span>
                </li>
                <li
                  class="ant-menu-item ant-menu-item-only-child"
                  data-menu-id="rc-menu-uuid-test-notes"
                  data-testid="side-menu"
                  role="menuitem"
                  style="padding-left: 24px;"
                  tabindex="-1"
                >
                  <span
                    class="ant-menu-title-content"
                  >
                    <div
                      class="text-center"
                    >
                      <div>
                        <svg
                          fill="white"
                        >
                          Notes.svg
                        </svg>
                      </div>
                      <p
                        class="ml-2 px-3 dl-mt-12 text-wrap text-center"
                      >
                        Add Notes
                      </p>
                    </div>
                  </span>
                </li>
              </ul>
              <div
                aria-hidden="true"
                style="display: none;"
              />
            </div>
          </aside>
          <div
            class="quickaction-component"
          >
            <h4
              class="dl-font-weight-500 dl-text-gray-900 text-center mb-5"
            >
              What’s next for you?
            </h4>
            <div>
              <div
                class="row action-con"
              >
                <div
                  class="col text-center"
                  data-testid="addtext-quickaction"
                >
                  <div
                    class="icon-con dl-pointer"
                  >
                    <svg>
                      Abc.svg
                    </svg>
                  </div>
                  <p
                    class="dl-pointer dl-text-gray-900 dl-mt-12"
                  >
                    Add Text
                  </p>
                </div>
                <div
                  class="col text-center"
                >
                  <div
                    class="icon-con dl-pointer "
                  >
                    <svg>
                      Template.svg
                    </svg>
                  </div>
                  <p
                    class="dl-pointer dl-text-gray-900 dl-mt-12"
                  >
                    Use Template
                  </p>
                </div>
              </div>
              <div
                class="row action-con"
              >
                <div
                  class="col text-center"
                >
                  <div
                    class="icon-con dl-pointer "
                  >
                    <svg>
                      UploadOutline.svg
                    </svg>
                  </div>
                  <p
                    class="dl-pointer dl-text-gray-900 dl-mt-12"
                  >
                    Upload design
                  </p>
                </div>
                <div
                  class="col text-center"
                >
                  <div
                    class="icon-con dl-pointer"
                  >
                    <svg>
                      Shirts.svg
                    </svg>
                  </div>
                  <p
                    class="dl-pointer dl-text-gray-900 dl-mt-12"
                  >
                    Change Products
                  </p>
                </div>
              </div>
            </div>
            <p
              class="dl-text-medium dl-text-gray-900 dl-font-weight-500 text-center"
            >
              or 
              <span
                class="dl-pointer dl-text-primary text-decoration-underline"
              >
                start over
              </span>
            </p>
          </div>
          <section
            class="ant-layout home-content dl-pt-12 dl-pb-12 dl-pl-12 dl-pr-12"
          >
            <main
              class="ant-layout-content d-flex"
            >
              <div
                class="history-action dl-pt-12 dl-pb-12 dl-pl-12 dl-pr-12 d-flex flex-column justify-content-between"
              >
                <div
                  class="text-center dl-pointer"
                >
                  <svg>
                    Undo.svg
                  </svg>
                  <p
                    class="dl-text-gray-600 dl-font-weight-500 mt-2"
                  >
                    Undo
                  </p>
                </div>
                <div
                  class="text-center opacity-25"
                >
                  <svg>
                    Redo.svg
                  </svg>
                  <p
                    class="dl-text-gray-600 dl-font-weight-500 mt-2"
                  >
                    Redo
                  </p>
                </div>
              </div>
              <div
                class="flex-grow-1 text-center"
              >
                <div
                  class="shirt-con d-flex justify-content-center"
                >
                  <div>
                    <div
                      class="konvajs-content"
                      role="presentation"
                      style="position: relative; user-select: none; width: 0px; height: 0px;"
                    >
                      <canvas
                        height="0"
                        style="padding: 0px; margin: 0px; border: 0px; background: transparent; position: absolute; top: 0px; left: 0px; width: 0px; height: 0px; display: block;"
                        width="0"
                      />
                      <canvas
                        height="0"
                        style="padding: 0px; margin: 0px; border: 0px; background: transparent; position: absolute; top: 0px; left: 0px; width: 0px; height: 0px; display: block;"
                        width="0"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="part-container"
                >
                  <div
                    class="part-item d-flex flex-column justify-content-center align-items-center part-item-active dl-pointer"
                  >
                    <svg>
                      ShirtFront.svg
                    </svg>
                    <p
                      class="dl-text-gray-600 dl-font-weight-600 dl-text-small text-uppercase dl-mt-8"
                    >
                      Front
                    </p>
                  </div>
                  <div
                    class="part-item d-flex flex-column justify-content-center align-items-center dl-pointer"
                  >
                    <svg>
                      ShirtBack.svg
                    </svg>
                    <p
                      class="dl-text-gray-600 dl-font-weight-600 dl-text-small text-uppercase dl-mt-8"
                    >
                      Back
                    </p>
                  </div>
                  <div
                    class="part-item d-flex flex-column justify-content-center align-items-center dl-pointer"
                  >
                    <svg>
                      RightSleeve.svg
                    </svg>
                    <p
                      class="dl-text-gray-600 dl-font-weight-600 dl-text-small text-uppercase dl-mt-8"
                    >
                      Right Sleeve
                    </p>
                  </div>
                  <div
                    class="part-item d-flex flex-column justify-content-center align-items-center dl-pointer"
                  >
                    <svg>
                      LeftSleeve.svg
                    </svg>
                    <p
                      class="dl-text-gray-600 dl-font-weight-600 dl-text-small text-uppercase dl-mt-8"
                    >
                      Left Sleeve
                    </p>
                  </div>
                </div>
                <div
                  class="d-flex justify-content-between dl-mt-8"
                >
                  <div
                    class="zoom-con d-flex justify-content-center align-items-center dl-pointer"
                  >
                    <svg>
                      ZoomOut.svg
                    </svg>
                  </div>
                  <div
                    class="zoom-con d-flex justify-content-center align-items-center dl-pointer"
                  >
                    <svg>
                      ZoomIn.svg
                    </svg>
                  </div>
                </div>
              </div>
            </main>
          </section>
        </div>
      </main>
    </section>
    <div
      class="position-fixed bottom-0 start-0 w-100 footer d-flex flex-row align-items-center justify-content-between dl-text-semi-small"
    >
      <div
        class="d-flex flex-row align-items-center"
      >
        <button
          class="dl-bg-primary-light dl-border-primary-light dl-btn border-1 dl-text-primary btn btn-secondary"
        >
          <svg>
            PlusCircle.svg
          </svg>
          <span
            class="ms-2 dl-text-semi-small"
          >
            Add Products
          </span>
        </button>
        <div
          class="small-shirt"
        >
          <img
            alt="shirt"
            height="64"
            src="ShirtSmall.png"
            width="56"
          />
        </div>
        <div
          class="vt-line dl-ml-16 dl-mr-16"
        />
        <div
          class="d-flex flex-column justify-content-bewtween"
        >
          <div
            class="d-flex flex-row align-items-end"
          >
            <p
              class="dl-text-medium dl-font-weight-400 dl-text-gray-900 dl-mr-8"
            >
              Gildan Unisex T-shirt
            </p>
            <p
              class="dl-text-semi-small dl-font-weight-500 dl-text-primary dl-pointer"
            >
              Change Product
            </p>
          </div>
          <div
            class="d-flex flex-row align-items-end mt-3"
          >
            <label
              class="ant-radio-wrapper"
            >
              <span
                class="ant-radio"
              >
                <input
                  class="ant-radio-input"
                  type="radio"
                  value=""
                />
                <span
                  class="ant-radio-inner"
                />
              </span>
              <span>
                <p
                  class="dl-text-semi-small dl-font-weight-400 dl-text-gray-900 dl-mr-8"
                >
                  White
                </p>
              </span>
            </label>
            <p
              class="dl-text-semi-small dl-font-weight-500 dl-text-primary dl-pointer"
            >
              Change Color
            </p>
          </div>
        </div>
      </div>
      <div
        class="d-flex flex-row align-items-center"
      >
        <button
          class="dl-btn dl-btn-outline dl-border-primary dl-text-primary dl-bg-white dl-mr-12 btn btn-secondary"
        >
          <svg>
            Share.svg
          </svg>
        </button>
        <button
          class="dl-bg-primary-light dl-border-primary-light dl-btn border-1 dl-text-primary dl-mr-12 d-flex flex-row align-items-center btn btn-secondary"
        >
          <svg>
            Save.svg
          </svg>
          <span
            class="ms-2 dl-text-semi-small"
          >
            Save
          </span>
        </button>
        <button
          class="dl-bg-primary dl-border-primary dl-btn border-1 dl-text-primary d-flex flex-row align-items-center btn btn-secondary"
        >
          <svg>
            Dollar.svg
          </svg>
          <span
            class="ms-2 dl-text-white dl-text-semi-small"
          >
            Get Pricing
          </span>
        </button>
      </div>
    </div>
  </section>
</div>
`;
